dvc/remote/ssh/connection.py

```diff
@@ -185,8 +185,23 @@ def download(self, src, dest, no_progress_bar=False, progress_title=None):
             self.sftp.get(src, dest, callback=pbar.update_to)
 
     def move(self, src, dst):
+        """Atomically move src to dst.
+
+        Moving is performed in two stages to make the whole operation atomic in
+        case src and dst are on different filesystems and actual physical
+        copying of data is happening.
+        """
         self.makedirs(posixpath.dirname(dst))
-        self.sftp.rename(src, dst)
+
+        tmp = tmp_fname(dst)
```

Comment:
is there something like `with tmp ...` to make sure we cleanup the stuff if there is some failure in the middle

Reply:
@shcheklein , we can do `finally` on the `try/except` block :thu

Reply:
yeah, I was just thinking that there might be something like `with tmp` already - since it makes a lot of sense and simplifies + deduplicates the code ... but finally works fine either

Reply:
@shcheklein , but what's the point of doing a two step operation if we are always removing the `tmp`? Isn't it useful to rollback the `src` when deleted or at least leave the `tmp`?

Reply:
not sure I got the question, @mroutis . My only point is that if something fails in the middle we don't want to stuck in some weird state (thus atomicity), and we don't want to leave some garbage after us that won't be controlled by anyone and can consume a lot of space (thus `with tmp ...` or `finally` or whatnot). `with smt ` is just a way of achieving the second requirement

Reply:
@shcheklein would you mind taking a look?